app/actions/websocket/roles.ts

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {fetchRolesIfNeeded, RolesRequest} from '@actions/remote/role';
+import {fetchMe} from '@actions/remote/user';
+import DatabaseManager from '@database/manager';
+import {queryRoleById} from '@queries/servers/role';
+import {queryCurrentUserId} from '@queries/servers/system';
+import {prepareMyTeams} from '@queries/servers/team';
+import {prepareUsers} from '@queries/servers/user';
+import {WebSocketMessage} from '@typings/api/websocket';
+
+import type {Model} from '@nozbe/watermelondb';
+import type {ServerDatabase} from '@typings/database/database';
+import type RoleModel from '@typings/database/models/servers/role';
+
+export async function handleRoleUpdatedEvent(serverUrl: string, msg: WebSocketMessage): Promise<void> {
+    const database = DatabaseManager.serverDatabases[serverUrl];
+    if (!database) {
+        return;
+    }
+
+    // only update Role records that exist in the Role Table
+    const role = JSON.parse(msg.data.role);
+    const dbRole = await queryRoleById(database.database, role.id);
+    if (!dbRole) {
+        return;
+    }
+
+    await database.database.write(async () => {
```

Comment:
Why not use directly `operator.handleRole`?

Reply:
The Role Table records are only a slice of the fields from the server API models. I also hadn't fully understood how the handleRole operator works.  Now I see that it doesn't uses a transformer to only include the desired fields from the API server models and excludes the rest. 

I've made this update. 

I do need to check the role record existance in the Role Table first though and exit early if the role doesn't already exist. The goal here is to only include roles applied to the current user and the roles are added when member and user roles updated.